hooks/useRbfTransactionData/helpers.ts

```diff
@@ -0,0 +1,121 @@
+  rbfTxSummary?: {
+    currentFee: number;
+    currentFeeRate: number;
+    minimumRbfFee: number;
+    minimumRbfFeeRate: number;
+  };
+  rbfRecommendedFees?: RbfRecommendedFees;
+  mempoolFees?: RecommendedFeeResponse;
+  isLoading?: boolean;
+  errorCode?: 'SOMETHING_WENT_WRONG';
+};
+
+export const isBtcTransaction = (
+  transaction: BtcTransactionData | StxTransactionData,
+): transaction is BtcTransactionData => transaction?.txType === 'bitcoin';
+
+export const constructRecommendedFees = (
+  lowerName: keyof RbfRecommendedFees,
+  lowerFeeRate: number,
+  higherName: keyof RbfRecommendedFees,
+  higherFeeRate: number,
+  stxAvailableBalance: string,
+): RbfRecommendedFees => {
+  const bigNumLowerFee = BigNumber(lowerFeeRate);
+  const bigNumHigherFee = BigNumber(higherFeeRate);
+
+  return {
+    [lowerName]: {
+      enoughFunds: bigNumLowerFee.lte(BigNumber(stxAvailableBalance)),
+      feeRate: microstacksToStx(bigNumLowerFee).toNumber(),
+      fee: microstacksToStx(bigNumLowerFee).toNumber(),
+    },
+    [higherName]: {
+      enoughFunds: bigNumHigherFee.lte(BigNumber(stxAvailableBalance)),
+      feeRate: microstacksToStx(bigNumHigherFee).toNumber(),
+      fee: microstacksToStx(bigNumHigherFee).toNumber(),
+    },
+  };
+};
+
+export const sortFees = (fees: RbfRecommendedFees) =>
+  Object.fromEntries(
+    Object.entries(fees).sort((a, b) => {
+      const priorityOrder = ['highest', 'higher', 'high', 'medium'];
+      return priorityOrder.indexOf(a[0]) - priorityOrder.indexOf(b[0]);
+    }),
+  );
+
+export const calculateStxData = async (
+  transaction: StxTransactionData,
+  btcNetwork: SettingsNetwork,
+  stacksNetwork: StacksNetwork,
+  appInfo: AppInfo | null,
+  stxAvailableBalance: string,
+): Promise<RbfData> => {
+  const { fee } = transaction;
+  const txRaw: string = await getRawTransaction(transaction.txid, btcNetwork);
+  const unsignedTx: StacksTransaction = deserializeTransaction(txRaw);
+
+  const [slow, medium, high] = await estimateTransaction(unsignedTx.payload, undefined, stacksNetwork);
+
+  let feePresets: RbfRecommendedFees = {};
+  let mediumFee = medium.fee;
+  let highFee = high.fee;
+  const higherFee = fee.multipliedBy(1.25).toNumber();
+  const highestFee = fee.multipliedBy(1.5).toNumber();
+
+  if (appInfo?.thresholdHighStacksFee) {
+    if (high.fee > appInfo.thresholdHighStacksFee) {
+      // adding a fee cap
+      highFee = appInfo.thresholdHighStacksFee * 1.5;
+      mediumFee = appInfo.thresholdHighStacksFee;
+    }
+  }
+
+  let minimumFee = fee.multipliedBy(1.25).toNumber();
+  if (!Number.isSafeInteger(minimumFee)) {
+    // round up the fee to the nearest integer
+    minimumFee = Math.ceil(minimumFee);
+  }
+
+  if (fee.lt(BigNumber(mediumFee))) {
+    feePresets = constructRecommendedFees('medium', mediumFee, 'high', highFee, stxAvailableBalance);
+  } else {
+    feePresets = constructRecommendedFees('higher', higherFee, 'highest', highestFee, stxAvailableBalance);
+  }
```

Comment:
nit: i always found better to define const to avoid any possible mistake while reassigning the var also can save you from if statements 

```ts
const shouldCapFee = appInfo?.thresholdHighStacksFee && high.fee > appInfo.thresholdHighStacksFee;

  const mediumFee = shouldCapFee ? appInfo.thresholdHighStacksFee : medium.fee;
  const highFee = shouldCapFee ? appInfo.thresholdHighStacksFee * 1.5 : high.fee;
  const higherFee = fee.multipliedBy(1.25).toNumber();
  const highestFee = fee.multipliedBy(1.5).toNumber();

  const defaultMinimumFee = fee.multipliedBy(1.25).toNumber();
  const minimumFee = !Number.isSafeInteger(defaultMinimumFee)
    ? Math.ceil(defaultMinimumFee)
    : fee.multipliedBy(1.25).toNumber();

  const feePresets: RbfRecommendedFees = fee.lt(BigNumber(mediumFee))
    ? constructRecommendedFees('medium', mediumFee, 'high', highFee, stxAvailableBalance)
    : constructRecommendedFees('higher', higherFee, 'highest', highestFee, stxAvailableBalance);
```

feel free to apply it, if it makes sense

Reply:
makes sense, thank you, I applied this change